tests/provision_test.py

```diff
@@ -0,0 +1,75 @@
+ # Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import os
+import unittest
+import subprocess
+from shlex import split
+import json
+
+from google.cloud.container_v1.services import cluster_manager
+
+class TestGKECluster(unittest.TestCase):
+    location = ''
+    name = ''
+
+    @classmethod
+    def setUpClass(cls):
+        """ Get the location of GKE cluster for later queries """
+        command = ('gcloud container clusters list --format="value(location)"')
+        result = subprocess.run(split(command), encoding='utf-8', capture_output=True)
+        cls.location = str(result.stdout.replace('\n', ''))
+        if cls.location == '':
+            raise Exception('GKE cluster is not provisioned!')
+        cls.name = 'projects/' + getProjectId() + '/locations/' + cls.location + '/clusters/stackdriver-sandbox'
+
+    def testNodeMachineType(self):
+        """" Test if the machine type for the nodes is as specified """
+        client = cluster_manager.ClusterManagerClient()
+        cluster = client.get_cluster(name=self.__class__.name)
+        node_config = cluster.node_config
+        self.assertEqual(node_config.machine_type, 'n1-standard-2')
+
+    def testNumberOfNode(self):
+        """" Test if the number of nodes in the node pool is in the specified range """
+        client = cluster_manager.ClusterManagerClient()
+        cluster = client.get_cluster(name=self.__class__.name)
+        self.assertTrue(3 < cluster.current_node_count < 10)
```

Comment:
This is not a definitive test.
We should assert on a specific number of nodes, not a range.

Reply:
There's an autoscaling in the config, so I guess the node number may not be definite in some circumstances?

Reply:
Good point. When running provisioning tests on CI, I don't expect loadgen to create synthetic traffic to the cluster, so autoscaling rules won't apply